### ISSUE TITLE
ClassVar in a dataclass transform is not detected when quoted

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#4004]
**Describe the bug**

When a `ClassVar` annotation is in quotes, pyright doesn't recognize it as such and reports an error if the argument is not provided at the construction site.

**To reproduce:**
```python
import typing as t
from typing_extensions import dataclass_transform

@dataclass_transform()
class Base:
    cv: "t.ClassVar[int]"


class Concrete(Base):
    cv = 6


c = Concrete()  # error: Argument missing for parameter "cv"
```
the following passes:
```python
@dataclass_transform()
class Base:
    cv: t.ClassVar[int]
```

**VS Code extension or command-line**
pyright 1.1.273, vscode extension shows the same problem

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.274, which I just published. It will also be included in a future release of pylance.
